scripts/build.sh

```diff
@@ -133,6 +133,9 @@ gem install cgi -v $CGI_RELEASE_VERSION -q --silent
 #Install rexml
 gem install rexml -v $REXML_RELEASE_VERSION -q --silent
 
+#Uninstall rexml 3.2.8 which comes with ruby 3.3.4, this is installed because of a dependency from the gem rss-0.3.0
+gem uninstall 'rexml:3.2.8'
```

Comment:
So this uninstall is unrelated to the previous line doing a `gem install` for rexml?

Reply:
Yep, the previous line adds a newer version of rexml but DD snags on the older version being there as well.  Ruby allows multiple versions of gems to be available.  Once the container is built, you can see both at:

```bash
gem list | grep rexml
rexml (3.3.6, 3.2.8)
```

After a manual removal you see:

```bash
gem uninstall 'rexml:3.2.8'
gem list | grep rexml
rexml (3.3.6)
```

I also tried just removing the gem before doing the install on line 134, that gave a warning message and I did not want to injure other gems:

```bash
gem uninstall rexml

You have requested to uninstall the gem:
        rexml-3.2.8

rss-0.3.0 depends on rexml (>= 0)
```

Reply:
The proposed solution returns a 0 exit code if the specific version of the gem is attempted to be removed more than once.

Reply:
I'm curious: do we need to explicitly include` rexml` in this container? It seems like a dependency that projects can add if they need it?

Looks like @dandersonsw added the line to install rexml in this PR: https://github.com/cloud-gov/general-task/pull/179/files

If we don't need it in the global container, could we just do one `gem uninstall rexml` and avoid the dependency mess?

Reply:
David added this since DD was detecting this older gem version being installed and was attempting to patch for it, not that a particular tool had it as a requirement.

The install of Ruby itself is installing 86 gems, many of those gems have dependencies on each other, like in this case.  In order to remove rexml we'd also need to remove rss-0.3.3 and then anything it is dependent on.  

This doesn't feel safe trying to make decisions on which 86 gems should be kept/removed for Ruby itself, we don't have the testing and knowledge of how ruby is used by every concourse job.